core/transaction_pool/impl/transaction_pool_impl.cpp

```diff
@@ -185,6 +199,17 @@ namespace kagome::transaction_pool {
       }
       removed.push_back(tx);
     }
+    // make sure that we don't revalidate extrinsics that were part of the
+    // recently imported block. This is especially important for UTXO-like
+    // chains cause the inputs are pruned so such transaction would go to future
+    // again.
+    for (auto &hash : known_imported_hashes) {
+      moderator_->ban(hash);
```

Comment:
not covered by tests

Reply:
I will reconsider the need of this part and then either remove or test it.